lib/timing.mjs

```diff
@@ -0,0 +1,135 @@
+import process from 'node:process';
+import { table } from 'table';
+
+// Inspired by ESLint's timing.js
+// https://github.com/eslint/eslint/blob/09bc2a88c00aa9a93c7de505795fc4e85b2e6357/lib/linter/timing.js
+
+/**
+ * Start time measurement.
+ * @returns {bigint} Variable for tracking time in nanoseconds.
+ */
+function startTime() {
+	return process.hrtime.bigint();
+}
+
+/**
+ * End time measurement.
+ * @param {bigint} start Variable for tracking time in nanoseconds.
+ * @returns {number} The measured time in milliseconds.
+ */
+function endTime(start) {
+	const diff = process.hrtime.bigint() - start;
+
+	return Number(diff) / 1e6;
+}
+
+const HEADERS = ['#', 'Rule', 'Time (ms)', 'Relative'];
+
+/**
+ * Decide how many rules to show in the output list.
+ * @returns {number} The number of rules to show.
+ */
+function getListSize() {
+	const TIMING = process.env.TIMING;
+
+	if (typeof TIMING === 'undefined') {
+		return 0;
+	}
+
+	if (TIMING.toLowerCase() === 'all') {
+		return Number.POSITIVE_INFINITY;
+	}
+
+	const parsed = Number.parseInt(TIMING, 10);
+
+	if (!isNaN(parsed) && parsed >= 1) {
```

Comment:
[suggestion] Minor. How about using `Number.isNaN()` instead of the global `isNaN()`? It's more robust, according to the [MDN document](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#difference_between_number.isnan_and_global_isnan).

```suggestion
	if (!Number.isNaN(parsed) && parsed >= 1) {
```

Reply:
Fixed it!
[Refactor applied suggestion](https://github.com/stylelint/stylelint/pull/8108/commits/6cfe6382bc03d70138000c8eaeac609628733eee)